vowpalwabbit/gd.cc

```diff
@@ -466,7 +466,8 @@ inline void pred_per_update_feature(norm_data& nd, float x, float& fw)
         }
         w[normalized] = x_abs;
       }
-      nd.norm_x += x2 / (w[normalized] * w[normalized]);
+      if (w[normalized] != 0)
```

Comment:
Do we know where this is coming from?  This is deep in a white-hot inner loop so it compromises computational integrity.